lib/logger.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file Abstracted logging properties and methods.
+ *
+ * If for any reason, they need a different logging tool or need to be extended,
+ * the changes can be made here.
+ */
+var util = require('util');
+var utility = require('./utility');
+var winston = require('winston');
+
+/**
+ * Function to get default options for an implementation of patternlab-import
+ * @return {Object}  options  an object of default patternlab-import options
+ */
+var getDefaultOptions = function (options) {
+  'use strict';
+
+  var options = {
+    logger: {
+      level: 'info',
+      silent: false,
+      exitOnError: true,
+      logFile: ''
+    }
+  };
+
+  return options;
+};
+
+/**
+ * Abstracts fully-fleshed logging module to the logger property of this module.
+ *
+ * If logging with metadata, or requiring winston.log's more advanced features,
+ * use exports.logger().log() instead of exports.log().
+ * @example
+ *   exports.logger(options).log('info', 'Test Log Message', { anything: 'This is metadata' });
+ *
+ * @param {Object} projectOptions - Configuration options.
+ * @return {Object} An instance of winston.Logger.
+ */
+exports.logger = function (projectOptions) {
```

Comment:
@scottnath I'm about to give up on this. I tried thinking of multiple ways of keeping a persistent instantiated transport object and it's not working. This is the closest I got:
```
exports.transportFile = new winston.transports.File({
  filename: options.logger.logFile,
  level: options.logger.level
});
```
But how do you pass the options object without this being a function? And if it's a function, how would you pass in an already instantiated object every time it was called?

Anyhow, I don't think this instantiation is all that expensive, but you see how we could run into problems with instantiations that are.